test/commands/submission/representation/generate_basic_test.rb

```diff
@@ -0,0 +1,164 @@
+require 'test_helper'
+
+class Submission::Representation::GenerateBasicTest < ActiveSupport::TestCase
+  test "creates submission representation record" do
+    filename = "something.rb"
+    code = "foo"
+    digest = Submission::Representation.digest_ast("#{filename}\n#{code}")
+    submission = create :submission
+    create :submission_file, submission:, filename:, content: code
+
+    uuid = SecureRandom.compact_uuid
+    SecureRandom.expects(:compact_uuid).returns(uuid).at_least_once
+
+    Submission::Representation::GenerateBasic.(submission)
+
+    representation = submission.reload.submission_representation
+
+    assert_equal submission, representation.submission
+    assert_equal "basic-#{uuid}", representation.tooling_job_id
+    assert_equal 200, representation.ops_status
+    assert_equal digest, representation.ast_digest
+  end
+
+  test "creates exercise representation" do
+    filename = "something.rb"
+    code = "foo"
+    normalised_code = "#{filename}\n#{code}"
+    digest = Submission::Representation.digest_ast(normalised_code)
+    submission = create :submission
+    create :submission_file, submission:, filename:, content: code
+
+    Submission::Representation::GenerateBasic.(submission)
+
+    perform_enqueued_jobs
+
+    assert_equal 1, Exercise::Representation.count
+    representation = Exercise::Representation.first
+
+    assert_equal submission.exercise, representation.exercise
+    assert_equal normalised_code, representation.ast
+    assert_equal digest, representation.ast_digest
+    assert_empty(representation.mapping)
+    assert_equal 0, representation.representer_version
+    assert_equal 1, representation.exercise_version
+    assert_equal 1, representation.num_submissions
+    assert_equal submission.created_at, representation.last_submitted_at
+  end
+
+  test "normalises ast correctly" do
+    filename_1 = "start.rb"
+    code_1 = "foo eeee"
+    filename_2 = "end.rb"
+    code_2 = "bar wwww \n aaaa"
+
+    # Check things are normalised correctly
+    digest = Submission::Representation.digest_ast("#{filename_1}\n#{code_1.gsub(/\s/,
+      '')}\n\n#{filename_2}\n#{code_2.gsub(/\s/, '')}")
```

Comment:
I wonder if using a plain multiline string without the interpolated values and gsubbing, which makes reading it a bit harder I feel:

```suggestion
# Check things are normalised correctly
    expected_ast = <<~AST
      start.rb
      fooeeee

      end.rb
      barwwwwaaaa
    AST
    digest = Submission::Representation.digest_ast(expected_ast.chomp)
```

Reply:
Yep.